File: php/36-valid-sudoku.php
<?php
class Solution {

    /**
     * @param String[][] $board
     * @return Boolean
     */
    function isValidSudoku($board) {
        $row = array_fill(0, 9, false);
        $col = $row;
        $cell = $row;
        for ($i = 0; $i < 9; $i++) {
            for ($j = 0; $j < 9; $j++) {
                if ($board[$i][$j] == '.') {
                    continue;
                }
                $c = $board[$i][$j] - 1;
                $cellN = 3 * intval($i / 3) + intval($j / 3);
                if ($row[$i][$c] || $col[$j][$c] || $cell[$cellN][$c]) {
                    return false;
                }
                $row[$i][$c] = true;
                $col[$j][$c] = true;
                $cell[$cellN][$c] = true;
            }
        }
        return true;
    }
}

File: go/36-valid-sudoku.go
package main

func isValidSudoku(board [][]byte) bool {
	r := len(board)
	c := len(board[0])
	row := make(map[int]map[int]bool, r)
	col := make(map[int]map[int]bool, c)
	cell := make(map[int]map[int]bool, 9)
	for i := 0; i < r; i++ {
		row[i] = make(map[int]bool, c)
		col[i] = make(map[int]bool, c)
		cell[i] = make(map[int]bool, 9)
	}
	for i := 0; i < r; i++ {
		for j := 0; j < c; j++ {
			if board[i][j] == '.' {
				continue
			}
			v := int(board[i][j]) - 1
			if exist, ok := row[i][v]; ok && exist {
				return false
			}
			if exist, ok := col[j][v]; ok && exist {
				return false
			}
			cellN := 3*(i/3) + (j / 3)
			if exist, ok := cell[cellN][v]; ok && exist {
				return false
			}
			row[i][v] = true
			col[j][v] = true
			cell[cellN][v] = true
		}
	}
	return true
}
